### DISCUSSION TITLE
Write multicall

### DISCUSSION THREAD

**Discussion by gershido** [wevm/viem#741] [Question] [answered]
Hey :) From the docs, it looks like `multicall` is available only from `publicClient`. Does a write `multicall` also supported from the `walletClient`?
Seems to me that the example on the docs (Contract section) doesn't make sense, as the `mint` function cannot happen from a `publicClient`. Thank you!

<img width="468" alt="Screenshot 2023-06-20 at 10 50 12" src="https://github.com/wagmi-dev/viem/assets/81111572/1f13d82f-582d-45cb-b075-ec4f0e0a97fa">

**Comment by jxom** [comment, marked as answer]
Ah yes, we should update that example to a more realistic one with a read function.

But no, write multicall doesn’t exist. Eventually we will support EIP-4337 which will have support for transaction bundlers.

**Reply by songkeys** [reply]
Hi. Is there a roadmap or plan on how EIP-4337 will be implemented?

**Comment by itsdiay** [comment]
Any updates on when write multicall will be supported?

**Comment by dvgui** [comment]
Is there another recommended way to use multicall? like with the ethers provider and another library? while this is being resolved

**Reply by jbrower95** [reply]
- https://github.com/mds1/multicall?tab=readme-ov-file#batch-contract-writes
   - note from their docs, multicall write is largely unsupported / "try at your own risk"

**Reply by dvgui** [reply]
it is not unsupported, it works perfectly fine if you know what you are doing and they warn developers accordingly. because there certainly are risks but also use cases when done correctly
